auem/data/datasets/audioset.py

```diff
@@ -90,6 +89,52 @@ def __next__(self):
         return d
 
 
+class AudiosetAnnotationReaderV2:
+    def __init__(self, annotation_path: Union[Path, str], classes: List):
+        self.annotation_path = annotation_path
+        self.classes = classes
+
+        self._annotations = None
+        self._idx = None
+
+    def _load_annotations(self):
+        self._annotations = []
+        with open(self.annotation_path, "r") as fh:
+            for l in csv.reader(
+                fh, quotechar='"', delimiter=",", skipinitialspace=True
+            ):
+                if not l[0].startswith("#"):
+                    self._annotations.append(l)
+
+    def __len__(self):
+        if not self._annotations:
+            self._load_annotations()
+
+        return len(self._annotations)
+
+    def __getitem__(self, idx):
+        if not self._annotations:
+            self._load_annotations()
+
+        item = self._annotations[idx]
+
+        return {
+            "ytid": item[0],
+            "start_seconds": float(item[1].strip()),
+            "end_seconds": float(item[2].strip()),
+            "classes": [self.classes.index(k) for k in item[3].split(",")],
```

Comment:
I guess this is a fundamental difference between the two different annotators. Doing it this way basically makes the annotation reader specialized for the only dataset and the task that dataset we've written up to this point. My reasoning about returning everything about the classes in a dictionary in V1 is separation of concerns. This way we'll be able to use the annotation reader in more places without having to rewrite it every time.

Reply:
I don't _exaclty_ disagree, however I think that taking that approach is too much premature optimization, whereas we just need to get the basic thing done right now; i'm not opposed to adding that back in in the future when it becomes necessary.